File: app/src/main/java/org/stepic/droid/core/presenters/contracts/SectionsView.java
package org.stepic.droid.core.presenters.contracts;

import org.stepic.droid.model.Section;

import java.util.List;

public interface SectionsView {

    void onEmptySections();

    void onConnectionProblem();

    void onNeedShowSections(List<Section> sectionList);

    void onLoading();

}

File: app/src/main/java/org/stepic/droid/core/presenters/contracts/SectionsView.kt
package org.stepic.droid.core.presenters.contracts

import org.stepic.droid.persistence.model.DownloadProgress
import org.stepik.android.model.Section

interface SectionsView {

    fun onEmptySections()

    fun onConnectionProblem()

    fun onNeedShowSections(sectionList: List<Section>)

    fun onLoading()

    fun updatePosition(position: Int)

    fun showDownloadProgress(progress: DownloadProgress)
}
